nvm.sh

```diff
@@ -2393,7 +2402,15 @@ nvm_install_source() {
   if ! (
     # shellcheck disable=SC2086
     command mkdir -p "${TMPDIR}" && \
-    command "${tar}" -x${tar_compression_flag}f "${TARBALL}" -C "${TMPDIR}" --strip-components 1 && \
+    if [ "${NVM_OS}" = 'openbsd' ]; then \
+      if [ "${tar_compression_flag}" = 'J' ]; then \
+        command xzcat "${TARBALL}" | "${tar}" -xf - -C "${TMPDIR}" -s '/[^\/]*\///'; \
+      else \
+        command "${tar}" -x${tar_compression_flag}f "${TARBALL}" -C "${TMPDIR}" -s '/[^\/]*\///'; \
+      fi; \
+    else \
+      command "${tar}" -x${tar_compression_flag}f "${TARBALL}" -C "${TMPDIR}" --strip-components 1; \
+    fi && \
```

Comment:
these three lines seem to be identical in both places. let's maybe make a new function instead that takes `IS_XZ`, `TARBALL`, and `TMPDIR` as arguments, and then does the right thing?

Reply:
PR has been updated to refactor the repeated lines (including a few additional tarball-extraction related lines) into a new function.